sdk/servicebus/service-bus/test/public/sessionsTests.spec.ts

```diff
@@ -411,3 +417,86 @@ describe.skip("SessionReceiver - disconnects", function(): void {
     refreshConnectionCalled.should.be.greaterThan(0, "refreshConnection was not called.");
   });
 });
+
+describe("SessionReceiver - disconnects", function(): void {
+  let serviceBusClient: ServiceBusClientForTests;
+  async function beforeEachTest(testClientType: TestClientType): Promise<EntityName> {
+    serviceBusClient = createServiceBusClientForTests();
+    return serviceBusClient.test.createTestEntities(testClientType);
+  }
+
+  after(() => {
+    return serviceBusClient.test.after();
+  });
+
+  it("calls processError and closes the link", async function(): Promise<void> {
+    const testMessage = TestMessage.getSessionSample();
+    // Create the sender and receiver.
+    const entityName = await beforeEachTest(TestClientType.UnpartitionedQueueWithSessions);
+    const receiver = await serviceBusClient.acceptSession(
+      entityName.queue!,
+      testMessage.sessionId,
+      {
+        maxAutoLockRenewalDurationInMs: 10000 // Lower this value so that test can complete in time.
+      }
+    );
+
+    let receiverSecondMessageResolver: Function;
+    let errorIsThrownResolver: Function;
+    const errorIsThrown = new Promise((resolve) => {
+      errorIsThrownResolver = resolve;
+    });
+    const receiverSecondMessage = new Promise((resolve) => {
+      receiverSecondMessageResolver = resolve;
+    });
+
+    const sender = serviceBusClient.createSender(entityName.queue!);
+    should.equal(receiver.isClosed, false, "Receiver should not have been closed");
+
+    // Send a message so we can be sure when the receiver is open and active.
+    await sender.sendMessages(testMessage);
+    receiver.subscribe(
+      {
+        async processMessage(_message: ServiceBusReceivedMessage) {
+          // Simulate a disconnect being called with a non-retryable error.
+          (receiver as any)["_context"].connection["_connection"].idle();
+        },
+        async processError(err) {
+          err.error.message && (err.error as ServiceBusError).code.should.equal("SessionLockLost");
+          errorIsThrownResolver();
+        }
+      },
+      { autoCompleteMessages: false }
+    );
+
+    await errorIsThrown;
+    should.equal(receiver.isClosed, true, "Receiver should have been closed");
+
+    const connectionContext = (receiver as any)["_context"];
+    const refreshConnection = connectionContext.refreshConnection;
+    let refreshConnectionCalled = 0;
+    connectionContext.refreshConnection = function(...args: any) {
```

Comment:
I was thinking to show the new receive(after the disconnect) would work fine on the same sbClient because the connection is refreshed.
This is just another validation that it indeed happens.

However, the new receive works anyway, so it should be evident by that. I guess I could remove the `refreshConnectionCalled` check then.